gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py

```diff
@@ -0,0 +1,183 @@
+import argparse
+import logging
+
+import hail as hl
+
+from gnomad.utils.slack import slack_notifications
+from gnomad.utils.sparse_mt import densify_sites
+from gnomad.utils.vcf import SPARSE_ENTRIES
+
+
+from gnomad_qc.v3.resources.basics import get_gnomad_v3_mt
+from gnomad_qc.v3.resources.release import release_sites
+from gnomad_qc.v3.resources.annotations import get_info, last_END_position
+from gnomad_qc.slack_creds import slack_token
+
+
+logging.basicConfig(
+    format="%(asctime)s (%(name)s %(lineno)s): %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger("get_impacted_variants")
+logger.setLevel(logging.INFO)
+
+
+def get_het_non_ref_impacted_var(
+    mt: hl.MatrixTable, info_ht: hl.Table, freq_ht: hl.Table
+) -> None:
+    """
+    Filter to variants where homalt hotfix incorrectly adjusts het nonref genotype calls.
+
+    :param hl.MatrixTable mt: Raw, split MatrixTable annotated with original genotype het nonref status.
+    :param hl.Table info_ht: Info Hail Table containing AS_lowqual information.
+    :param hl.Table freq_ht: Hail Table containing 455K frequency information.
+    :return: None
+    """
+
+    logger.info("Removing low qual variants and star alleles...")
+    mt = mt.filter_rows(
+        (~info_ht[mt.row_key].AS_lowqual)
+        & ((hl.len(mt.alleles) > 1) & (mt.alleles[1] != "*"))
+    )
+
+    logger.info("Filtering to common (AF > 0.01) variants...")
+    mt = mt.annotate_rows(AF=freq_ht[mt.row_key].freq[0].AF)
+    mt = mt.filter_rows(mt.AF > 0.01)
+
+    logger.info(
+        "Filtering to variants with at least one het nonref call and checkpointing..."
+    )
+    mt = mt.filter_rows(hl.agg.sum(mt.het_non_ref) >= 1)
+
+    logger.info(
+        "Adding new genotype entries with original homalt hotfix and het nonref-aware hotfix..."
+    )
+    mt = mt.annotate_entries(
```

Comment:
Haven't modified this yet, but when I make it a real PR let me know if this is what you are thining

Reply:
yeah, I think your way of filtering makes more sense than the extra `call_stats` calls. it might be easier to follow if the code runs like this:

- Filter to impacted variants using `mt = mt.filter_rows(hl.agg.any(mt.het_non_ref & ((mt.AD[1] / mt.DP) > 0.9)))`
- Adjust genotypes using code from `GT_hetnonref_fix`